velox/dwio/common/DirectDecoder.h

```diff
@@ -92,7 +94,24 @@ class DirectDecoder : public IntDecoder<isSigned> {
       } else if constexpr (std::is_same_v<
                                typename Visitor::DataType,
                                int128_t>) {
-        toSkip = visitor.process(super::template readInt<int128_t>(), atEnd);
+        if (precision_.has_value()) {
```

Comment:
This does not look correct.  As far as I see you should not need to change anything at decoder level (the int96 implementation is a little hacky and you should follow it), just use it as the plain vanilla int64_t decoder, and all these conversion should happen inside `TimestampColumnReader` (or rename it to something else to avoid name clash with the one reading int96).

Reply:
I also have PR for timestamp filter support for INT64. So I did try moving the conversion inside `TimestampColumnReader` but it was causing issue with timestamp filter. As the filtering was happening with int64 values in decoder. I was getting `testInt64() is not supported.` Maybe I missed something.

Let me wait for the INT96 pr to go in first (since that is almost ready), as some effort is common in both to avoid conflicts

Reply:
You can process the filter in column reader (there is no way to handle it in int decoder, different format has different timestamp representation), see example in https://github.com/facebookincubator/velox/blob/09e1b0d0e77738bd6974922c60e921906c41942d/velox/dwio/dwrf/reader/SelectiveTimestampColumnReader.cpp#L228

Reply:
Updated change based on the suggestion. 
Will try to merge both classes INT64, INT96 readers if feasible.